projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/BgpAdvertisement.java

```diff
@@ -423,27 +424,25 @@ public int getWeight() {
 
   @Override
   public int hashCode() {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + _asPath.hashCode();
-    result = prime * result + _clusterList.hashCode();
-    result = prime * result + _communities.hashCode();
-    result = prime * result + _dstIp.hashCode();
-    result = prime * result + _dstNode.hashCode();
-    result = prime * result + (_dstVrf != null ? _dstVrf.hashCode() : 0);
-    result = prime * result + _localPreference;
-    result = prime * result + Long.hashCode(_med);
-    result = prime * result + _network.hashCode();
-    result = prime * result + _nextHopIp.hashCode();
-    result = prime * result + _originType.ordinal();
-    result = prime * result + _originatorIp.hashCode();
-    result = prime * result + _srcIp.hashCode();
-    result = prime * result + _srcNode.hashCode();
-    result = prime * result + _srcProtocol.ordinal();
-    result = prime * result + _srcVrf.hashCode();
-    result = prime * result + _type.ordinal();
-    result = prime * result + _weight;
-    return result;
+    return Objects.hash(
+        _asPath,
+        _clusterList,
+        _communities,
+        _dstIp,
+        _dstNode,
+        _dstVrf,
+        _localPreference,
+        _med,
+        _network,
+        _nextHopIp,
+        _originType,
+        _originatorIp,
+        _srcIp,
+        _srcNode,
+        _srcProtocol,
```

Comment:
Does Objects.hash of an Enum use Enum.hashCode or Enum.ordinal? I think to be safe you should just hash the ordinal. Using hash of Enum itself instead of its ordinal was the cause inconsistent refs for BDP (and other issues I believe) earlier.

Reply:
> hash ... cause of inconsistent refs

Can you say more? Hash codes are not guaranteed to be the same across different implementations of JVM, versions of JVM, or even different instances of the JVM.

Nothing should not rely on hash code consistency.

Reply:
We may need to remove dependence on hash code consistency in the future. However, for BDP currently, it is required.
All the types that are hashed in BDP at the moment are deterministic by specification (that is, we do not rely on `Object.hashCode`). This might actually not matter for this class at the moment if it is not hashed during BDP oscillation detection, so perhaps this change is not currently required. But it is a simple change nonetheless.